src/Stratis.Bitcoin.Features.Consensus/Behaviors/ProvenHeadersConsensusManagerBehavior.cs

```diff
@@ -123,12 +122,18 @@ protected override Payload ConstructHeadersPayload(GetHeadersPayload getHeadersP
                 return null;
             }
 
-            var headers = new ProvenHeadersPayload();
-            foreach (ChainedHeader header in this.chain.EnumerateToTip(fork).Skip(1))
+            var provenHeadersPayload = new ProvenHeadersPayload();
+
+            var chainedHeaderToStartFrom = DetermineFirstHeaderToConstructPayloadFrom(fork);
```

Comment:
I suggest, for readability, this approach.
```
ChainedHeader header = this.FindHeaderToStartFrom(frok)
``` 
And delete `chainedHeaderToStartFrom`